ships/0027-image-abstraction-layer.md

```diff
@@ -0,0 +1,233 @@
+| from           | Keeps a reference from where the image got imported |
+| importedAt     | Date and time of the import |
+| imageReference | Where this reference points to (by hash), may point to the mirror registry in case of a mirrored Image |
+
+### ImageImport status
+
+On a ImageImport status, one would be able to find information about the import attempts executed
+by the operator, it looks like:
+
+| Name           | Description  |
+| -------------- | ------------ |
+| hashReference  | Once the import has succeeded this is populated with the data about the image. Properties for hashReference are equal to Image.Status.HashReferences above. This is what is copied into the Image.Status.HashReferences list. |
+| condition      | The current condition for the ImageImport expressed as a single metav1.Condition{} |
+| importAttempts | A list of all attempts made at processing the ImageImport |
+
+The properties for the object `.status.hashReference` are exactly the same as for the objects
+inside `Image.Status.HashReferences`. As for `status.importAttempts` objects they contain the
+following properties:
+
+| Name    | Description                                                                          |
+| ------- | ------------------------------------------------------------------------------------ |
+| when    | Date and time of the import attempt                                                  |
+| succeed | A boolean indicating if the import was successful or not                             |
+| reason  | In case of failure (succeed = false), what was the error                             |
+
+The operator must try to import multiple times the same ImageImport in case of failures. These
+imports should implement exponential backoff to avoid “too many requests” from remote registries.
+After ten failures the operator should stop trying to process the ImageImport.
+
+
+### User Stories
+
+As an operator I want to be able to create a reference by hash to an image hosted in a remote
+registry so I can easily refer to it by its hash instead of by tag.
+
+As an operator I want to be able to mirror an image in my registry so I can take ownership of it
+and possibly gain some performance while pulling.
+
+
+### Implementation Notes
+
+To import images hosted in registries protected by authentication the operator would need to be
+able to read docker credentials secrets that exist in the same namespace where the Image exists.
+The idea is to mimic OpenShift’s Image Stream here: read all docker config secrets from the target
+namespace and use the appropriate one (if present).
+
+### Test Plan
+
+As with the Build APIs today, this operator should include thorough and high-quality unit tests,
+integration tests, and end-to-end tests. End to end tests are to be designed using kuttl utility.
+
+### Release Criteria
+
+Not currently targeting a release.
+
+
+### Risks and Mitigations
+
+**Confusion with OpenShift’s ImageStream implementation**
```

Comment:
In my view this EP clearly explains the the relationship, `image` is inspired but meant to work against any Kubernetes cluster, like all other Shipwright components. Also, we focus on the features that are most important to this organization, instead of trying to pursue a drop-in replacement.

Reply:
IIUC we would not need to worry about this naming confusion the, is that what you mean?